docs/releases.md

```diff
@@ -1,39 +1,157 @@
+
+We might expand the six week release schedule in case of:
+- No new updates to be added
+- In case of a large community event that takes the core team availability away (e.g. IPFS Conf, Dev Meetings, IPFS Camp, etc)
+
+## Release Flow
+
+`go-ipfs` releases come in 4 stages:
+
+- **Release Stage 1 - Internal testing** - Test the Release against our testing infrastructure, including interoperability, integration, test lab, multiple runtimes and the apps we've built (WebUI, Desktop, NPM on IPFS, HTTP Client Libraries). The intent is to make this stage fully automated (and somewhat is already), until then, we manually check a list and ensure that all tests have been run
+- **Release Stage 2 - Invite IPFS partners to test** - Reach out to our partners (i.e. projects that have volunteered to support `go-ipfs` by using their own test infrastructure and tell us the results)
+- **Release Stage 3 - Announce to the broader community** - Communicate to the community that a new Release Candidate is ready and that everyone is welcome to test it with us
+- **Release Stage 4 - Complete the Release** - Finalize the release, start the next release.
+
+The Release Stages are not linked to Release Candidate numbers, in fact, there can be multiple release candidate per stages as we catch bugs and improve the release itself.
+
+<p align="center">
+  <a href="https://ipfs.io">
+    <img src="https://gateway.ipfs.io/ipfs/QmaFtLxoCAm5vFQ9AftKkhJwSAdDdF1jzV9DfzW6gbXqFL/Paper.Sketches.23.png" width="450" />
+  </a>
+</p>
+
+## Performing a Release
+
+The first step is for the `Lead Maintainer` for `go-ipfs` to open an issue with Title `go-ipfs <version> Release` and a c&p of the following template:
+
+```
+> <short tl;dr; of the release>
+
+# 🗺 What's left for release
+
+<List of items with PRs and/or Issues to be considered for this release>
+
+# 🔦 Highlights
+
+<List of items with PRs and/or Issues to be considered for this release>
+
+# 🏗 API Changes
+
+<List of API changes, if any>
+
+# ✅ Release Checklist
+
+For each RC published in each stage:
 - [ ] version string in `version.go` has been updated
 - [ ] tag commit with vX.Y.Z-rcN
 
-## Pre-Release Checklist
-- [ ] before release, tag 'release candidate' for users to test against
-  - if bugs are found/fixed, do another release candidate
-- [ ] all tests pass (no exceptions)
-- [ ] run interop tests https://github.com/ipfs/interop#test-with-a-non-yet-released-version-of-go-ipfs
-- [ ] webui works (for most definitions of 'works') - Test the multiple pages and verify that no visible errors are shown.
+### Release Stage 1 - Internal testing
```

Comment:
Note: as noted on Release Flow, this whole stage can potentially be automated. Until it is, let's have the items as a check list so that we don't forget any :)